src/systems/interactions.js

```diff
@@ -0,0 +1,162 @@
+      const manifoldPoint = persistentManifold.getContactPoint(j);
+      if (manifoldPoint.getDistance() <= 10e-6) {
+        return findHandCollisionTarget(driver.els.get(handPtr === body0ptr ? body1ptr : body0ptr).object3D);
+      }
+    }
+  }
+}
+
+const remoteHoverTargets = new Map();
+function findRemoteHoverTarget(o) {
+  if (!o) return null;
+  const target = remoteHoverTargets.get(o.uuid);
+  return target || findRemoteHoverTarget(o.parent);
+}
+AFRAME.registerComponent("is-remote-hover-target", {
+  init: function() {
+    remoteHoverTargets.set(this.el.object3D.uuid, this.el);
+  }
+});
+
+AFRAME.registerSystem("interaction", {
+  updateCursorIntersection: function(intersection) {
+    this.rightRemoteHoverTarget = intersection && findRemoteHoverTarget(intersection.object);
+  },
+
+  init: function() {
+    this.options = {
+      leftHand: {
+        entity: document.querySelector("#player-left-controller"),
+        grabPath: paths.actions.leftHand.grab,
+        dropPath: paths.actions.leftHand.drop,
+        constraintTag: "offersHandConstraint",
+        hoverFn: findHandCollisionTargetForHand
+      },
+      rightHand: {
+        entity: document.querySelector("#player-right-controller"),
+        grabPath: paths.actions.rightHand.grab,
+        dropPath: paths.actions.rightHand.drop,
+        constraintTag: "offersHandConstraint",
+        hoverFn: findHandCollisionTargetForHand
+      },
+      rightRemote: {
+        entity: document.querySelector("#cursor"),
+        grabPath: paths.actions.cursor.grab,
+        dropPath: paths.actions.cursor.drop,
+        constraintTag: "offersRemoteConstraint",
+        hoverFn: this.getRightRemoteHoverTarget
+      }
+    };
+    this.state = {
+      leftHand: {
+        hovered: null,
+        held: null,
+        spawning: null
+      },
+      rightHand: {
+        hovered: null,
+        held: null,
+        spawning: null
+      },
+      rightRemote: {
+        hovered: null,
+        held: null,
+        spawning: null
+      }
+    };
+
+    this.cursorController = document.querySelector("#cursor-controller");
+  },
+
+  getRightRemoteHoverTarget() {
+    return this.rightRemoteHoverTarget;
+  },
+
+  tickInteractor(options, state) {
+    const userinput = AFRAME.scenes[0].systems.userinput;
+    if (state.held) {
+      const networked = state.held.components["networked"];
+      const lostOwnership = networked && networked.data && networked.data.owner !== NAF.clientId;
+      if (userinput.get(options.dropPath) || lostOwnership) {
+        state.held = null;
+      }
+    } else {
+      state.hovered = options.hoverFn.call(this, options.entity.body);
+      if (
+        state.hovered &&
+        state.hovered.components.tags &&
+        state.hovered.components.tags.data.isHoldable &&
+        userinput.get(options.grabPath) &&
+        (this.el.is("frozen") || !state.hovered.components.pinnable || !state.hovered.components.pinnable.data.pinned)
```

Comment:
Conceptually I like the idea that `isHoldable` should just be made to be false or true depending on whether we actually want it to be holdable or not this frame. That approach seems a bit tricky to get right, since it depends on the state's frozen state, the object's pinnable state, and whether the object was `isHoldable` to begin with, so for now I'm going to do this check here. As soon as another part of the app wants to check the `isHoldable` tag for some reason, let's revisit this decision.

As for breaking up the boolean, I'm not convinced that buys us anything.